pkg/apis/extensions/validation/validation.go

```diff
@@ -564,3 +569,30 @@ func ValidateClusterAutoscaler(autoscaler *extensions.ClusterAutoscaler) errs.Va
 	allErrs = append(allErrs, validateClusterAutoscalerSpec(autoscaler.Spec)...)
 	return allErrs
 }
+
+func ValidatePodSelector(ps extensions.PodSelector) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateLabels(ps.MatchLabels, "matchLabels")...)
+	for i, expr := range ps.MatchExpressions {
```

Comment:
The only case I can think of is the one I mentioned: exists and notin. This is actually the behavior I originally expected from notin. An alternative to allowing multiple requirements would be to create a new notin operator (say, !in) with this behavior. (!in would be slightly tricky to parse, but is pithy.)

We couldn't enforce non-repeated keys on list/watch filters, but could enforce it for uses of PodSelector in the API.